lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1419,6 +1420,10 @@ def generate_students_certificates(
     statuses_to_regenerate = task_input.get('statuses_to_regenerate', [])
     students_require_certs = students_require_certificate(course_id, enrolled_students, statuses_to_regenerate)
 
+    if statuses_to_regenerate:
```

Comment:
@saleem-latif I think you should `invalidate_generated_certificates ` before getting list of student require certificate by calling `students_require_certificate ` and revert you changes in `students_require_certificate` which you made earlier. That would make things less complicated.